kbcstorage/workspaces.py

```diff
@@ -164,10 +165,14 @@ def load_files(self, workspace_id, file_mapping, preserve=None):
             raise Exception('Loading files to workspace is only available for ABS workspaces')
         files = Files(self.root_url, self.token)
         file_list = files.list(tags=file_mapping['tags'])
-        inputs = {}
+        jobs = Jobs(self.root_url, self.token)
         for file in file_list:
-            inputs[file['id']] = file_mapping['destination']
-        body = _make_body(inputs, source_key='dataFileId')
-        body['preserve'] = preserve
-        url = '{}/{}/load'.format(self.base_url, workspace['id'])
-        return self._post(url, data=body)
+            inputs = {
+                file['id']: "%s/%s" % (file_mapping['destination'], file['name'])
+            }
+            body = _make_body(inputs, source_key='dataFileId')
+            # always preserve the workspace, otherwise it would be silly
+            body['preserve'] = 1
+            url = '{}/{}/load'.format(self.base_url, workspace['id'])
+            job = self._post(url, data=body)
+            jobs.block_until_completed(job['id'])
```

Comment:
can we have it like that it stacks the jobids in an array and then waits until all of them are completed?

```
for job_id in job_ids:
    jobs.block_until_completed(job_id)
```

Reply:
sure, but I don't really see what difference it makes

Reply:
I guess your way it would still load the ones that didn't fail

Reply:
the difference is that when you load multiple files it either:
- posts and loads them sequentially (you post another file only until the previous one is finished) or
- post them sequentially and loads them in parallel

Reply:
ah, right, that too!

Reply:
whether it's success or error makes imho no difference here

Reply:
which is probably a good point - what happens when the load_files fail?
e.g. you attempt to load a non-existent file

Reply:
right, it will complete regardless 🤦 .  

Yeah, I'll put in some failure warning too

Reply:
not sure how to test that though since it'll only attempt to load files that it found the matching tag for

Reply:
hm, would invalid destination trigger it?

Reply:
I will try that

Reply:
So unfortunately not, because it doesn't get a chance to create the job, it fails validation when creating it